bin/download-logs

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../lib/maze'
+require_relative '../lib/maze/browser_stack_utils'
+require_relative '../lib/maze/logger'
+require 'net/http'
+require 'json'
+require 'uri'
+require 'optimist'
+require 'fileutils'
+
+class DownloadLogEntry
+  def start(args)
+    p = Optimist::Parser.new do
+      text 'Download device sessions logs from BrowserStack using the build ID'
+      text ''
+      text 'Requires BROWSER_STACK_USERNAME and BROWSER_STACK_ACCESS_KEY'
+      text ''
+      text 'Usage [OPTIONS] <filenames>'
+      text ''
+      opt :help,
+          'Print this help.'
+      opt :build_id,
+          "ID of the BrowserStack build",
+          :type => :string
+    end
+
+    opts = Optimist::with_standard_exception_handling p do
+      raise Optimist::HelpNeeded if ARGV.empty? # show help screen
+      p.parse ARGV
+    end
+
+    # Get browserstack username and access key from the environment
+    username = ENV['BROWSER_STACK_USERNAME']
+    access_key = ENV['BROWSER_STACK_ACCESS_KEY']
+
+    # Check if BROWSER_STACK_USERNAME or BROWSER_STACK_ACCESS_KEY has been set
+    if username.nil? || access_key.nil?
+      $logger.warn "BROWSER_STACK_USERNAME or BROWSER_STACK_ACCESS_KEY has not been set"
+      Optimist::with_standard_exception_handling p do
+        raise Optimist::HelpNeeded
+      end
+    end
+
+
+    build_info = Maze::BrowserStackUtils.build_info username,
+                                                    access_key,
+                                                    opts[:build_id]
+
+    $logger.info "Getting logs for build: #{opts[:build_id]}"
+
+    build_info.each_with_index do |session, index|
+      $logger.info "Downloading Device Logs for Session: #{index + 1} to #{Dir.pwd}/maze_output"
+      Maze::BrowserStackUtils.download_log username,
+                                           access_key,
+                                           "#{opts[:build_id]}-#{index + 1}",
```

Comment:
Assigning this to a local variable first (and using on L64 too) would make it more readable overall.